docs/program/apis/data-client.md

```diff
@@ -0,0 +1,246 @@
+
+**Returns**:
+
+- [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): ID of the new file.
+
+```python {class="line-numbers linkable-line-numbers"}
+time_requested = datetime(2023, 6, 5, 11)
+time_received = datetime(2023, 6, 5, 11, 0, 3)
+
+file_id = await data_client.tabular_data_capture_upload(
+    part_id="INSERT YOUR PART ID",
+    component_type='motor',
+    component_name='left_motor',
+    method_name='IsPowered',
+    tags=["tag_1", "tag_2"],
+    data_request_times=[(time_requested, time_received)],
+    tabular_data=[{'PowerPCT': 0, 'IsPowered': False}]
+)
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/app/data_client/index.html#viam.app.data_client.DataClient.tabular_data_capture_upload).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### FileUpload
+
+Upload arbitrary files stored on your machine to the [Viam app](https://app.viam.com) by file name.
+Uploaded files can be found under the **Files** subtab of the app's [**Data** tab](https://app.viam.com/data).
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `part_id` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Part ID of the component used to capture the data. See [Find Part ID](#find-part-id) for instructions on retrieving this value.
+- `component_type` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Type of the component used to capture the data.
+- `component_name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Name of the component used to capture the data.
+- `method_name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Name of the method used to capture the data.
+- `file_name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Optional name of the file. The empty string `""` will be assigned as the filename if one isn’t provided.
+- `file_extension` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Optional file extension. The empty string `""` will be assigned as the file extension if one isn’t provided.
+- `tags` [(Optional[List[str]])](https://docs.python.org/3/library/stdtypes.html#typesseq-list): Optional list of [image tags](/manage/data/label/#image-tags) to allow for tag-based data filtering when retrieving data.
+- `data` [(Optional[bytes])](https://docs.python.org/3/library/stdtypes.html#bytes-objects): Optional bytes representing file data to upload.
+
+**Returns**:
+
+- [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): ID of the new file.
+
+```python {class="line-numbers linkable-line-numbers"}
+file_id = await data_client.file_upload(
+    part_id="INSERT YOUR PART ID",
+    component_type='motor',
+    component_name='left_motor',
```

Comment:
I would shy a bit away from including component name/type (and remove method_name in general), since we generally want to steer people to associating files with their robot part.

Reply:
@agreenb Updated! LMK if correct/what you were looking for. I removed "method_name" as a parameter on both file upload methods and also removed the component fields from the sample code